requests_mv_integrations/request_mv_integration.py

```diff
@@ -0,0 +1,1774 @@
+            verbose: (optional) Boolean, provide request and response details.
+
+        Returns:
+            requests.Response
+
+        """
+        tune_request = TuneRequest(
+            retry_tries=self.retry_tries,
+            retry_backoff=self.retry_backoff,
+            retry_codes=self.request_retry_http_status_codes
+        )
+
+        self.session = tune_request.session
+
+        self.logger.debug(
+            "Send Request: Details: {}".format(
+                request_label if request_label else ""
+            ),
+            extra={
+                'request_method': request_method,
+                'request_url': request_url,
+                'request_params': request_params,
+                'request_data': request_data,
+                'request_json': request_json,
+                'request_headers': request_headers,
+                'request_auth': request_auth,
+                'request_session': request_session,
+                'cookie_payload': cookie_payload,
+                'timeout': timeout,
+                'allow_redirects': allow_redirects,
+                'verify': verify,
+                'stream': stream,
+                'verbose': verbose
+            }
+        )
+
+        if not request_method:
+            raise TuneRequestError(
+                error_message="Parameter 'request_method' not defined"
+            )
+        if not request_url:
+            raise TuneRequestError(
+                error_message="Parameter 'request_url' not defined"
+            )
+
+        self.built_request_curl = None
+
+        if request_session:
+            self.logger.debug(
+                "Send Request: Session: Requested"
+            )
+
+            if self.session is None:
+                self.logger.debug(
+                    "Send Request: Session: New"
+                )
+                self.session = requests.Session()
+
+            self.logger.debug(
+                "Send Request: Session: Existing",
+                extra={
+                    'cookie_payload': self.session.cookies.get_dict(),
+                    'request_label': request_label
+                }
+            )
+
+        response = None
+        headers = None
+
+        if request_headers:
+            headers = request_headers
+
+        request_method = request_method.upper()
+
+        if request_data and isinstance(request_data, str):
+            if len(request_data) <= 20:
+                request_data_extra = request_data
+            else:
+                request_data_extra = request_data[:20] + ' ...'
+        else:
+            request_data_extra = safe_str(request_data)
+
+        request_extra = {
+            'request_method': request_method,
+            'request_url': request_url,
+            'timeout': timeout,
+            'request_params': safe_dict(request_params),
+            'request_data': request_data_extra,
+            'request_headers': safe_dict(headers)
+        }
+
+        if verbose:
```

Comment:
We shouldn't be using passing flags around to switch the log level. The 'verbose' flag should be removed from our codebase and log levels set appropriately.